static/app/views/issueList/utils/getIssueTagValues.tsx

```diff
@@ -1,29 +1,94 @@
  */
-export function makeGetIssueTagValues(
-  tagValueLoader: (key: string, search: string) => Promise<TagValue[]>
-) {
+export function makeGetIssueTagValues({
+  organization,
+  pageFilters,
+  useCache = true,
+}: FetchIssueTagValuesParams): (tag: Tag, query: string) => Promise<string[]> {
   return async (tag: Tag, query: string): Promise<string[]> => {
+    const orgSlug = organization.slug;
+    const projectIds = pageFilters.projects.map(id => id.toString());
+    const endpointParams = {
+      start: pageFilters.datetime.start
+        ? getUtcDateString(pageFilters.datetime.start)
+        : undefined,
+      end: pageFilters.datetime.end
+        ? getUtcDateString(pageFilters.datetime.end)
+        : undefined,
+      statsPeriod: pageFilters.datetime.period,
+    };
+    const {data: eventsTagValues = []} = useFetchOrganizationTagKeyValues(
+      {
+        orgSlug,
+        tagKey: tag.key,
+        search: query,
+        dataset: Dataset.ERRORS,
+        useCache: useCache,
+        projectIds,
+        ...endpointParams,
+      },
+      {}
+    );
+
+    const {data: issuePlatformTagValues = []} = useFetchOrganizationTagKeyValues(
+      {
+        orgSlug,
+        tagKey: tag.key,
+        search: query,
+        dataset: Dataset.ISSUE_PLATFORM,
+        useCache: useCache,
+        projectIds,
+        ...endpointParams,
+      },
+      {}
+    );
+
     // device.class is stored as "numbers" in snuba, but we want to suggest high, medium,
     // and low search filter values because discover maps device.class to these values.
     if (isDeviceClass(tag.key)) {
-      return DEVICE_CLASS_TAG_VALUES;
+      return await Promise.resolve(DEVICE_CLASS_TAG_VALUES);
     }
-    const values = await tagValueLoader(tag.key, query);
-    return values.map(({value}) => {
-      // Truncate results to 5000 characters to avoid exceeding the max url query length
-      // The message attribute for example can be 8192 characters.
-      if (typeof value === 'string' && value.length > 5000) {
-        return value.substring(0, 5000);
+
+    const allTagValuesDict: Record<string, TagValue> = {};
+    eventsTagValues.forEach(value => {
+      allTagValuesDict[value.name] = value;
+    });
+
+    issuePlatformTagValues.forEach(value => {
+      if (allTagValuesDict[value.name]) {
+        allTagValuesDict[value.name].count += value.count;
       }
-      return value;
+      allTagValuesDict[value.name] = value;
     });
+
+    const allTagValues = Object.values(allTagValuesDict);
+
+    return await Promise.resolve(
+      allTagValues.map(({value}) => {
+        // Truncate results to 5000 characters to avoid exceeding the max url query length
+        // The message attribute for example can be 8192 characters.
+        if (typeof value === 'string' && value.length > 5000) {
+          return value.substring(0, 5000);
+        }
+        return value;
+      })
+    );
```

Comment:
This `await Promise.resolve(...)` is pretty janky, but is a workaround to force the return type of this function to be `Promise<string[]>`, which is what both the new and old search components expect in their `getTagValue` prop. In the future when the old search component is completely phased out, we should refactor this and get rid of this jankyness, but I don't think it'll cause much harm until then.